src/main/java/org/opensearch/security/dlic/rest/api/AccountApiAction.java

```diff
@@ -128,12 +146,22 @@ protected void handleGet(RestChannel channel, RestRequest request, Client client
         try {
             builder.startObject();
             final User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+            final String username = user.getName();
+            final SecurityDynamicConfiguration<?> internalUser = load(CType.INTERNALUSERS, false);
             if (user != null) {
                 final TransportAddress remoteAddress = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_REMOTE_ADDRESS);
                 final Set<String> securityRoles = privilegesEvaluator.mapRoles(user, remoteAddress);
                 final SecurityDynamicConfiguration<?> configuration = load(getConfigName(), false);
 
-                builder.field("user_name", user.getName())
+               if (request.path().endsWith(SAVED_TENANT)){
+                    if (configuration.exists(user.getName())){
+                        InternalUserV7 iu = (InternalUserV7) internalUser.getCEntry(username);
+                        builder.field("saved_tenant", iu.getSaved_tenant());
+                    } else {
+                        builder.field("message", "Sorry, saved tenant currently only stored for internal users.");
```

Comment:
Would it be better like this? 
"Sorry, saved tenant currently only stored for internal users." -> "Sorry, saved tenant currently **are** only stored for **existing** internal users."

Reply:
Makes sense; will be changed.